packages/@aws-cdk/aws-amplify/lib/branch.ts

```diff
@@ -161,3 +198,76 @@ export class Branch extends Resource implements IBranch {
     return this;
   }
 }
+
+class AmplifyAssetDeploymentProvider extends NestedStack {
+  /**
+   * Returns the singleton provider.
+   */
+  public static getOrCreate(scope: Construct) {
+    const providerId =
+      'com.amazonaws.cdk.custom-resources.amplify-asset-deployment-provider';
+    const stack = Stack.of(scope);
+    const group =
+      (stack.node.tryFindChild(providerId) as AmplifyAssetDeploymentProvider) ?? new AmplifyAssetDeploymentProvider(stack, providerId);
+    return group.provider.serviceToken;
+  }
+
+  private readonly provider: Provider;
+
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+
+    const onEvent = new NodejsFunction(
```

Comment:
Are we sure we want to use a `NodejsFunction` here? @MrArnoldPalmer usually custom resources included in the CDK use normal Lambda functions...

Reply:
For the dependency bundling + building I feel like it's nice to have. Do you think there are significant downsides? I suppose build times of the repo could be affected. Anything else though?

My feeling is we should be okay layering abstractions here to make managing custom resource handlers easier, especially since NodeJSFunction is stable.

Reply:
Discussed with @skinny85 and he agreed it was okay. It's main affects will be on user's synth time who are including this in their apps, as if they don't have `esbuild` installed globally, the docker build could take some time. That being said, it's an implementation detail that we can change in the future if we so desire.